components/Header/index.jsx

```diff
@@ -0,0 +1,45 @@
+import { Flex, Spacer, Box, Text, Heading, Button, Link } from '@chakra-ui/react';
+import { useRouter } from 'next/dist/client/router';
+
+const Header = ({ brandName, }) => {
+    const { push } = useRouter();
+
+    const onClickLoginHandler = () => {
+        push('/auth/login')
+    }
+    const onClickRegistrationHandler = () => {
+        push('/auth/registration')
+    }
+    const onClickLogoHandler = () => {
+        push('/')
+    }
```

Comment:
These functions can create a new ref each time which causes an extra rendering for a child in which they are passing as props, I would suggest wrapping them in `useCallack`.